pricenode/src/main/java/bisq/price/spot/ExchangeRateProvider.java

```diff
@@ -59,21 +62,55 @@
  */
 public abstract class ExchangeRateProvider extends PriceProvider<Set<ExchangeRate>> {
 
-    public static final Set<String> SUPPORTED_CRYPTO_CURRENCIES = CurrencyUtil.getAllSortedCryptoCurrencies().stream()
-            .map(TradeCurrency::getCode)
-            .collect(Collectors.toSet());
-
-    public static final Set<String> SUPPORTED_FIAT_CURRENCIES = CurrencyUtil.getAllSortedFiatCurrencies().stream()
-            .map(TradeCurrency::getCode)
-            .collect(Collectors.toSet());
-
+    private static Set<String> SUPPORTED_CRYPTO_CURRENCIES = new HashSet<>();
+    private static Set<String> SUPPORTED_FIAT_CURRENCIES = new HashSet<>();
     private final String name;
     private final String prefix;
+    private final Environment env;
 
-    public ExchangeRateProvider(String name, String prefix, Duration refreshInterval) {
+    public ExchangeRateProvider(Environment env, String name, String prefix, Duration refreshInterval) {
         super(refreshInterval);
         this.name = name;
         this.prefix = prefix;
+        this.env = env;
+    }
+
+    public Set<String> getSupportedFiatCurrencies() {
+        if (SUPPORTED_FIAT_CURRENCIES.size() == 0) {         // one-time initialization
```

Comment:
`.isEmpty()` is more expressive, same in line 84 `.filter(ccy -> !ccy.isEmpty())` and the crypto currency method